tests/test_processor.py

```diff
@@ -119,6 +118,29 @@ def loop():
     assert len(t.events) == 3  # nothing should have been executed
 
 
+async def test_reminder_cancelled(default_processor):
+    out = CollectingOutputChannel()
+    sender_id = uuid.uuid4().hex
+
+    d = Dispatcher(sender_id, out, default_processor.nlg)
+    r = ReminderScheduled("utter_greet", datetime.datetime.now(),
+                          kill_on_user_message=True)
+    t = default_processor.tracker_store.get_or_create_tracker(sender_id)
+
+    t.update(UserUttered("test"))
+    t.update(ActionExecuted("action_reminder_reminder"))
+    t.update(r)
+    t.update(ReminderCancelled("utter_greet"))
+
+    default_processor.tracker_store.save(t)
+    await default_processor.handle_reminder(r, d)
+
+    # retrieve the updated tracker
+    t = default_processor.tracker_store.retrieve(sender_id)
+    # there should be no utter_greet action
+    assert ActionExecuted("utter_greet") not in t.events
```

Comment:
I think we should also check the jobs of the scheduler here making sure it is not scheduled